source/opt/local_single_store_elim_pass.cpp

```diff
@@ -319,6 +120,127 @@ void LocalSingleStoreElimPass::InitExtensions() {
       "SPV_EXT_descriptor_indexing",
   });
 }
+bool LocalSingleStoreElimPass::ProcessVariable(ir::Instruction* var_inst) {
+  vector<ir::Instruction*> users;
+  FindUses(var_inst, &users);
+
+  ir::Instruction* store_inst = FindSingleStoreAndCheckUses(var_inst, users);
+
+  if (store_inst == nullptr) {
+    return false;
+  }
+
+  return RewriteLoads(store_inst, users);
+}
+
+ir::Instruction* LocalSingleStoreElimPass::FindSingleStoreAndCheckUses(
+    ir::Instruction* var_inst, const vector<ir::Instruction*>& users) const {
+  // Make sure there is exactly 1 store.
+  ir::Instruction* store_inst = nullptr;
+
+  // If |var_inst| has an initializer, then that will count as a store.
+  if (var_inst->NumInOperands() > 1) {
+    store_inst = var_inst;
+  }
+
+  for (ir::Instruction* user : users) {
+    switch (user->opcode()) {
+      case SpvOpStore:
+        if (store_inst == nullptr) {
+          store_inst = user;
+        } else {
+          // More than 1 store.
+          store_inst = nullptr;
```

Comment:
This is incorrect and causes bad code to be generated. This logic just forgets that a store was seen. One way to correct this is to replace this with "return nullptr;".

Reply:
That use to an `if-then-else` instead of the `switch`.  So the break broke out of the loop.  It also was originally in a larger part of the code instead of being in its own function do a return would not have worked.  Then I made changes and do not have a test for this.  Good catch.  I'll try to create a test case for this.  You suggestion should work.